sound/soc/sof/topology.c

```diff
@@ -1233,6 +1243,60 @@ static int sof_control_unload(struct snd_soc_component *scomp,
  * DAI Topology
  */
 
+static int sof_core_enable(struct snd_sof_dev *sdev, int core)
+{
+	struct sof_ipc_pm_core_config pm_core_config = {
+		.hdr = {
+			.cmd = SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_CORE_ENABLE,
+			.size = sizeof(pm_core_config),
+		},
+		.enable_mask = sdev->enabled_cores_mask | BIT(core),
+	};
+	int ret;
+
+	if (sdev->enabled_cores_mask & BIT(core))
+		return 0;
+
+	/* power up the core */
+	ret = snd_sof_dsp_core_power_up(sdev, BIT(core));
```

Comment:
this doesn't scale to e.g. TGL, where only core0 can be powered-up by the driver. Other cores will have to be powered with an IPC sent to core0.

Reply:
@plbossart this piece of code exists even in the code today but out of curiosity, what happens when the driver tries to power up one of the salve cores this way?

Reply:
no idea. timeout or nothing, either way it's not going to be successful.

Reply:
@plbossart @RanderWang AFAIK the TGL milti-core support is still in the works

Reply:
@lyakh  Now the default DSP frequency is in low mode, how about to set it in high mode ? I just got this message from @keyonjie  and don't know how to change it.

Reply:
@RanderWang sorry, I'm not sure where that frequency is configured either and how is it related to this PR?

Reply:
@RanderWang You need to check the firmware Kconfig and disable the config for low frequency clock, and rebuild the firmware.

Reply:
Full TGL support is beyond this patch (right @plbossart), but do we need some checks to this patch so that it does not try to do illegal actions on TGL (only core0 should be powered by driver). @lyakh ?

Reply:
I have a conceptual issue with a patch to support multi-core that doesn't address a known architecture. That essentially means we are going to undo-redo what we could have done right in the first place with more thinking.

It could very well be that the _power_up() function hides the actual enablement and dependency on IPC, but I'd like at least to see that this patch is compatible with directions for TGL. Ignoring TGL is not an option.

Reply:
@plbossart my understanding is, that multi-core on TGL isn't yet supported by the firmware, see e.g. https://github.com/thesofproject/sof/pull/2516 and as @ranj063 noted, this patch doesn't change the way SOF currently works or doesn't work om TGL specifically. As long as the topology doesn't specify cores other than 0, nothing will happen.

Reply:
@lyakh my understanding is that TGL multi-core support was a development that was supposed to happen very soon. @lbetlej @mmaka1 am I missing something here?

Reply:
@plbossart  yes, its on read-to-merge stage so please review kernel support for TGL.
it will not work If in TGL  snd_sof_dsp_ops .core_power_up = hda_dsp_enable_core

Reply:
@abonislawski sorry, has it been merged by now and I've forgotten or have you got a link to that PR?

Reply:
@lyakh not merged yet
https://github.com/thesofproject/sof/pull/2516